### ISSUE TITLE
[Android] Geolocation - Custom setInterval and setFastestInterval

### ISSUE THREAD

**Issue by tripodsgames** [ionic-team/capacitor-plugins#538]
### Plugin

Geolocation

### Description

Add an option to set a custom Interval and FastestInterval.

https://github.com/ionic-team/capacitor-plugins/blob/561fea2f27b54354ddac11ab7aa74b9b3f09084d/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/Geolocation.java#L52

https://github.com/ionic-team/capacitor-plugins/blob/561fea2f27b54354ddac11ab7aa74b9b3f09084d/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/Geolocation.java#L53

### Platform(s)

Android

### Preferred Solution

Add the options interval and fastestInterval in the PositionOptions interface.

**Comment by szundi** [comment]
Hello @jcesarmobile!

I think this should be handled as a bug, please consider the following reasoning.

Capacitor should provide one unified codebase for different platforms. With this code that is now in the Geolocation plugin here and there, Android now gives positions every 5 seconds instead of 1 seconds if it is specified in the timeout parameter. Web and iOS works well with the same user code and producing the desired output based on the API definition: positions every second.

I think according to the meaning of the "timeout" parameter, the code up there should not use 10000 and 5000 as constants, but the timeout parameter values for setInterval and setFastestInterval too - the only way it makes sure the app works according to the API on all platforms, including Android. This is the result of the simpler API of this only "timeout" parameter.

Do you think the API "timeout" parameter actually means the requested position period?

Thanks for considering, Capacitor is awesome.

András

**Comment by szundi** [comment]
I can indeed confirm that when I change the 10000 and 5000 to the "timeout" variable value (1000 in my case), it works as specified in the Capacitor API, and the coordinates are refreshed at the correct period.

It is therefore seems to be a good solution to fix this bug, also it confirms that using these literals in the Android specific code are completely ignores the fact that there is this timeout parameter and this is the direct cause of the bug.

**Comment by kencam** [comment]
I can also confirm this. I changed setInterval to 1000, setFastestInterval to 500 and timeout to 1000 and it is working normally now.

**Comment by CoooWeee** [comment]
@tripodsgames I agree about the missing feature. But **fastestInterval** is deprecated. I would suggest bumping **playServicesLocationVersion** to '21.0.1' and using **LocationRequest.Builder**

Here the code:
```
                LocationRequest locationRequest = new LocationRequest.Builder(priority, timeout)
                        .setMinUpdateIntervalMillis(minUpdateInterval) // use this instead of fastestInterval
                        .setMaxUpdateDelayMillis(timeout)
                        .build();
```  

In regards to **setMaxUpdateDelayMillis** I am not too sure. **setWaitForAccurateLocation** is by default _true_ but it did seem as if it does not work as long as **setMaxUpdateDelayMillis** is low. I did understand the documentation differently.


I did test it and it works in my case.

**Comment by szundi** [comment]
@CoooWeee really thanks for spending the effort on replying, will try and report back with the results.

**Comment by tripodsgames** [comment]
> @tripodsgames I agree about the missing feature. But **fastestInterval** is deprecated. I would suggest bumping **playServicesLocationVersion** to '21.0.1' and using **LocationRequest.Builder**
> 
> Here the code:
> 
> ```
>                 LocationRequest locationRequest = new LocationRequest.Builder(priority, timeout) // instead of setPriority & setInterval
>                         .setMinUpdateIntervalMillis(minUpdateInterval) // instead of fastestInterval
>                         .setMaxUpdateDelayMillis(timeout) // instead of setMaxWaitTime
>                         .build();
> ```

I opened this issue a long time ago, I'll update with your suggestion

**Comment by pirex360** [comment]
Any news about this? for real-time tracking with fast movement...this is a pain, can you explain better where to put the code above?

**Comment by pirex360** [comment]
Ok, I get it. It's on Geolocation.java file in the capacitor plugin folder/geolocation

I changed to this, but it doesn't get the same results as google maps... it's not smoother to watch the location, it takes about 1 second to retrieve...and the position jumps a lot, I tried to play around with the values...it is quicker now...but still away of optimum results

`                LocationRequest locationRequest = new LocationRequest.Builder(1000)
                    .setMaxUpdateDelayMillis(100)
                    .setMinUpdateIntervalMillis(50)
                    .setPriority(priority)
                    .setFastestInterval(100);
                    .setInterval(1000);
                    .build();`
